### PR TITLE
Fix Multispeed heat pump resistive defrost bug - GitHub Issue 4563

### LINKED ISSUE #4563: Coil:Heating:DX:MultiSpeed Defrost Bug
from @jonwinkler,

I encountered a potential bug in EnergyPlus 8.2. If you diff the two attached IDFs, you’ll see they’re identical with the exception of the defrost strategy implemented by the heat pump. The file giving me the error is using Resistive defrost, which throws the following error:

```
** Severe  ** SizeDXCoil Coil:Heating:DX:MultiSpeed DX HEATING COIL
   **   ~~~   ** ... DataConstantUsedForSizing and DataFractionUsedForSizing used for autocalculating Speed 1 Gross Rated Heating COP [W] must both be greater than 0.
```
 
Changing the defrost strategy to `ReverseCycle` fixes the error. The cause of the error was challenging to track down because we don’t use autosizing in our simulations.

### PR DESCRIPTION
**Pull request by lgu1234**
Bugfix for 4563. Fixes #4563

### CODE REVIEW

File: tst/EnergyPlus/unit/DXCoils.Unit.cc [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	PerfCurve( CurveNum ).Coeff5 = 0.0;
+	PerfCurve( CurveNum ).Coeff6 = 0.0;
+	PerfCurve( CurveNum ).Var1Min = -100.0;
+	PerfCurve( CurveNum ).Var1Max = 100.0;
+	PerfCurve( CurveNum ).Var2Min = -100.0;
+	PerfCurve( CurveNum ).Var2Max = 100.0;
+
+	SetPredefinedTables();
+	SizeDXCoil( 2 );
+	EXPECT_EQ( 5000.0, DXCoil( 2 ).DefrostCapacity );
```
Comment (nmerket) [inline tst/EnergyPlus/unit/DXCoils.Unit.cc]: I had someone point this out to me on my last pull request, but we should be using the `EXPECT_DOUBLE_EQ` function when comparing floating point numbers in these unit tests. See [Floating-Point Comparison](https://code.google.com/p/googletest/wiki/V1_7_AdvancedGuide#Floating-Point_Comparison) in the Google test documenation.

Comment (lgu1234) [inline tst/EnergyPlus/unit/DXCoils.Unit.cc]: @nmerket I don't have a problem to make a change.
@Myoldmopar Do you agree it?

In addition, I only changed a line of code. How to make tests failed? Any idea?

**Comment by nmerket** [comment]
@lgu1234 I think these tests are failing because you haven't merged `develop` into your branch in quite a while, so your code is quite a bit different from what it is being compared to.

**Comment by lgu1234** [comment]
@nmerket or @Myoldmopar How can I do it?

**Comment by nmerket** [comment]
Just did it. Let's wait and see how the regression tests turn out.

**Comment by lgu1234** [comment]
Noel:

 

Thanks.

 

Gu

**Comment by lgu1234** [comment]
@nmerket I checked regression test results. They seems OK. Thanks.

**Comment by nmerket** [comment]
I just checked the results from the defect file and we're not hitting the error anymore. I also ran your unit test (after changing to `EXPECT_DOUBLE_EQ`) by running `energyplus_tests` and it passed. I think we're ready to merge.

@Myoldmopar I still can't get his unit test to show up when I run `ctest`. I'd feel better about merging this if I knew why. When you get a chance can you help me figure that out?

**Comment by Myoldmopar** [comment]
I think running it via energyplus_tests is the right/only way to run the unit test.  I don't think the individual gtests get populated to be called by ctest directly.  @kbenne ?

**Comment by kbenne** [comment]
All of the gtests are wrapped by ctest and can be run individually.  This is primarily to provide a consistent interface for automated testing, but it can also be convenient for humans.  The ctest command provides an interface similar to the gtest runner, to select tests, run tests in parallel (-j), regex filter, etc.  From the build directory type ctest --help for all of the options.  

Here I pass two options -I to only run tests 1-10 and -N to list the tests, but not run them.

```
kbenne-27067s:build kbenne$ ctest -I 1,10 -N
Test project /Users/kbenne/Development/EnergyPlus/build
  Test  #1: BackspaceTest.BackspaceEmptyIStream
  Test  #2: BackspaceTest.BackspaceIStream
  Test  #3: bitTest.Transfer
  Test  #4: bitTest.Basic
  Test  #5: bitTest.Size
  Test  #6: bitTest.Set
  Test  #7: bitTest.Clr
  Test  #8: bitTest.Test
  Test  #9: bitTest.Not
  Test #10: bitTest.And
```

If you wrote some gtests but they are not showing up in ctest it is because you need to manually rerun cmake.  The test files are grepped by cmake for ```add_tests``` macros and that is how ctest is populated with in individual tests.  There are no build rules in place to detect changes to test files and rerun cmake automatically as part of the build process.

**Pull request merged**